kernel/GBEngine/kstd2.cc

```diff
@@ -3563,6 +3572,21 @@ poly kNF2 (ideal F,ideal Q,poly q,kStrategy strat, int lazyReduce)
   assume(strat->R==NULL);//omfree(strat->R);
   omfree(strat->S_2_R);
   omfree(strat->fromQ);
+#ifdef HAVE_SHIFTBBA
+  // only LM of elements in S is shifted
+  // necessary to prevent deleting the tail multiple times
+  if (strat->tailRing->isLPring && strat->Shdl != NULL && strat->Shdl->m != NULL)
```

Comment:
Shdl is an ideal, i.e. the test should be:
if(rIsLPRing(currRing) && strat->Shdl!=NULL))